Userland/Libraries/LibAudio/FlacLoader.h

```diff
@@ -10,23 +10,27 @@
 #include "FlacTypes.h"
 #include "Loader.h"
 #include <AK/BitStream.h>
+#include <AK/Buffered.h>
 #include <AK/Stream.h>
 #include <AK/Types.h>
 #include <AK/Variant.h>
 #include <LibCore/FileStream.h>
 
 namespace Audio {
 
-class FlacInputStream : public Variant<Core::InputFileStream, InputMemoryStream> {
+class FlacInputStream : public Variant<Buffered<Core::InputFileStream>, InputMemoryStream> {
 
 public:
-    using Variant<Core::InputFileStream, InputMemoryStream>::Variant;
+    using Variant<Buffered<Core::InputFileStream>, InputMemoryStream>::Variant;
 
     bool seek(size_t pos)
     {
         return this->visit(
-            [&](Core::InputFileStream& stream) {
-                return stream.seek(pos);
+            [&](Buffered<Core::InputFileStream>& buffered) {
+                // Discard the buffer, then seek normally.
+                if (!buffered.discard_or_error(buffered.buffered()))
+                    return false;
+                return buffered.leak_stream().seek(pos);
```

Comment:
Feels like maybe this "discard and seek" could be wrapped in a member fn on Buffered rather than adding a method to reach into it's internals and possibly violate it's invariants. 

Does raise the question though, why do we want to discard the buffer on seek? What if the input stream is destructive (I.e. read once)

Reply:
If the stream can't be seeked, then `leak_stream` won't return something that has the `seek` method anyways. This is just a special case of a template instantiation where we know the Buffered underlying type to be seekable. So at least with the current IO stream architecture, it doesn't make sense to have this on the Buffered itself, but I had this thought as well. If @sin-ack gets his rearchitecting done, maybe we can put that in with something like `requires(IsSubclassOf<StreamT, Seekable>)`.